lib/myMap.ml

```diff
@@ -47,11 +47,23 @@ module type S = sig
 
 (* Bind keys to list of values *)
   val accumulate : key -> 'a -> 'a list t -> 'a list t
+
+  (* Compatibility with OCaml<5.1 *)
+  val of_list : (key * 'a) list -> 'a t
 end
 
 module Make(O:Set.OrderedType) : S with type key = O.t =
   struct
-    include Map.Make(O)
+    module M = Map.Make(O)
+
+    (* Compatibility with OCaml<5.1.
+       Because the one in 5.1 is significantly faster than this one, we let the
+       standard library version override this implementation. *)
```

Comment:
This is slightly surprising, because the code is identical. However having the same code _inside_ the body of the `Map,Make` functor may favor some optimisations, such as inlining.